src/main/kotlin/io/github/techbox/core/Techbox.kt

```diff
@@ -111,7 +127,7 @@ class Techbox(private val config: Config) {
             shardStartListener.latch = CountDownLatch(latchCount)
             this.shardManager = shardManager.build()
 
-            threadPool.submit {
+            launch {
```

Comment:
is important to know where the errors within the code come from during debugging.
Would be ideal to put a name for this coroutine, you can use it like this:
```kotlin
launch(CoroutineName("some name here")) {
   // ..
}
```